### ISSUE TITLE
Which layer of Google's NASNet should I use for extracting features for attention?

### ISSUE THREAD

**Issue by aayushARM** [DeepRNN/image_captioning#12]
I am implementing a network similar to this one, but want to use the pre-trained CNN with max accuracy over 2012ILSVRC dataset, i.e., [NASNet-large](https://github.com/tensorflow/models/tree/master/research/slim/nets/nasnet). Usually, people go by extracting the last convolution layer features. But NASNet's architecture is relatively complex and I couldn't find a direct Conv layer. Below is the Tensorboard visualization of the "final_layer" cell of NASNet:

![tb1](https://user-images.githubusercontent.com/20727454/36257604-911d7fb2-127d-11e8-9c25-68246d7883e5.png)

And below is the second last cell:

![tb2](https://user-images.githubusercontent.com/20727454/36257616-a1bb98ae-127d-11e8-8c77-13c4d8977801.png)

To me, the relu node I've selected in first image([1,11,11,4032]) seems close to what's needed for attention, but I am not sure. Any help will be highly appreciated.

**Comment by aayushARM** [comment]
I have currently decided to go with the relu node after taking some suggestions from others, hence closing this issue for now. Will post here later about updates.
